### PR TITLE
Feature db 150998 new rules

### PR DESCRIPTION
**Pull request by xe21500**

### CODE REVIEW

File: packages/eslint-plugin-bbva/docs/rules/no-transition-abort-in-routes.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  model(params, transition) {
+    abortIfNotFirst(transition);
+  }
+});
+```
+
+## Known limitations
+
+This rule only checks for certain hooks (`beforeModel`, `willTransition`, `loading`, `model`, `afterModel`, `redirect`, `error`, `resetController`) in files which might contain Ember route logic (files named `route.js` or whose path contains `routes` or `mixins`), but the active transition can always be accessed by retrieving the internal routing service state.
```
Comment (adrigzr) [inline packages/eslint-plugin-bbva/docs/rules/no-transition-abort-in-routes.md]: It is only needed on `beforeModel`, `model`, `afterModel` and `redirect` i think.

File: packages/eslint-plugin-bbva/docs/rules/prefer-throw-in-promise.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+foo().catch(() => {
+  return reject();
+});
+
+try {
+  return reject();
+} catch (e) {
+  // ...
+}
```
Comment (adrigzr) [inline packages/eslint-plugin-bbva/docs/rules/prefer-throw-in-promise.md]: This should only be incorrect when it is inside an async function.

File: packages/eslint-plugin-bbva/docs/rules/prefer-throw-in-promise.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+try {
+  // ...
+} catch (e) {
+  return reject();
```
Comment (adrigzr) [inline packages/eslint-plugin-bbva/docs/rules/prefer-throw-in-promise.md]: This should only be incorrect when it is inside an async function.

File: packages/eslint-plugin-bbva/lib/rules/no-throw-in-unchained-promise.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+		},
+		fixable: null,
+		schema: []
+	},
+
+	create(context) {
+		return {
+			// Matches any `throw` inside a `then` / `catch` that is not assigned to a variable or returned
+			[':not(:matches(ReturnStatement, VariableDeclarator, AssignmentExpression)) >' +
+			'CallExpression:matches([callee.property.name=then], [callee.property.name=catch]) ThrowStatement'](node) {
```
Comment (adrigzr) [inline packages/eslint-plugin-bbva/lib/rules/no-throw-in-unchained-promise.js]: Brackets are not needed here.

File: packages/eslint-plugin-bbva/lib/rules/no-transition-abort-in-routes.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+			recommended: false
+		},
+		fixable: null,
+		schema: [],
+	},
+
+	create(context) {
+		const filename = context.getFilename();
+
+		if (filename.endsWith('route.js') || filename.includes('routes') || filename.includes('mixins')) {
```
Comment (adrigzr) [inline packages/eslint-plugin-bbva/lib/rules/no-transition-abort-in-routes.js]: Maybe `reopens`?

File: packages/eslint-plugin-bbva/lib/rules/no-throw-in-unchained-promise.js (hunk 4 above)
Comment (xe21500) [reply packages/eslint-plugin-bbva/lib/rules/no-throw-in-unchained-promise.js]: Yep, it is, notice there is a `+` operator joining the two strings. I didn't like the multiline approach using template literals, as the final string would contain the tab characters. Any other formatting suggestions are very welcome here :)

File: packages/eslint-plugin-bbva/lib/rules/no-transition-abort-in-routes.js (hunk 5 above)
Comment (xe21500) [reply packages/eslint-plugin-bbva/lib/rules/no-transition-abort-in-routes.js]: Done

File: packages/eslint-plugin-bbva/docs/rules/prefer-throw-in-promise.md (hunk 2 above)
Comment (xe21500) [reply packages/eslint-plugin-bbva/docs/rules/prefer-throw-in-promise.md]: Yep, I agree, thanks!

File: packages/eslint-plugin-bbva/docs/rules/prefer-throw-in-promise.md (hunk 3 above)
Comment (xe21500) [reply packages/eslint-plugin-bbva/docs/rules/prefer-throw-in-promise.md]: Yep, I agree, thanks!

File: packages/eslint-plugin-bbva/docs/rules/empty-line-before-return.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+# Empty line before return statements (empty-line-before-return)
```
[the remark this answers is not in the document]
Comment (xe21500) [reply packages/eslint-plugin-bbva/docs/rules/empty-line-before-return.md]: In no way whatsoever considering I'm going to remove it and configure https://eslint.org/docs/rules/padding-line-between-statements instead 😅

**Pull request merged**